src/deluge/gui/menu_item/arpeggiator/midi_cv/note_probability.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2014-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "gui/menu_item/integer.h"
+#include "gui/ui/sound_editor.h"
+#include "model/clip/instrument_clip.h"
+#include "model/song/song.h"
+
+namespace deluge::gui::menu_item::arpeggiator::midi_cv {
+class NoteProbability final : public Integer {
+public:
+	using Integer::Integer;
+	void readCurrentValue() override {
+		auto* current_clip = getCurrentInstrumentClip();
+		int64_t value = (int64_t)current_clip->arpeggiatorNoteProbability;
+		this->setValue((value * kMaxMenuValue + 2147483648) >> 32);
+	}
+	void writeCurrentValue() override {
+		getCurrentInstrumentClip()->arpeggiatorNoteProbability = (uint32_t)this->getValue() * 85899345;
```

Comment:
where did this number come from? can it be extracted to a constexpr definition somewhere for arpeggiatorNoteProbability stuff?

Reply:
there is a comment in value_scaling.cpp that might explain it:
// (2147483648 / kMidMenuValue) == 85899345
If you search the code you will see it being used in several places. If I recall correctly, this was to scale the value somehow. I can't remember
<img width="557" alt="image" src="https://github.com/user-attachments/assets/1b1452d1-b1e1-440a-b1ac-a9e1da94fd16">

int32_t computeFinalValueForStandardMenuItem(int32_t value) {
	if (value == kMaxMenuValue) {
		return 2147483647;
	}
	else if (value == kMinMenuValue) {
		return -2147483648;
	}
	else {
		// (2147483648 / kMidMenuValue) == 85899345
		return (uint32_t)value * 85899345 - 2147483648;
	}
}

Reply:
Ok I found the refactor made with value_scaling and now I applied the same refactor to this parameter